arangod/GeneralServer/H2CommTask.cpp

```diff
@@ -397,14 +398,55 @@ bool H2CommTask<T>::readCallback(asio_ns::error_code ec) {
 
   doWrite();
 
-  if (!_writing && shouldStop()) {
+  if (!this->_writing && shouldStop()) {
     this->close();
     return false;  // stop read loop
   }
 
   return true;  //  continue read lopp
 }
 
+template <SocketType T>
+void H2CommTask<T>::setIOTimeout() {
+  double secs = GeneralServerFeature::keepAliveTimeout();
+  if (secs <= 0) {
+    return;
+  }
+  
+  const bool wasReading = this->_reading;
+  const bool wasWriting = this->_writing;
+  TRI_ASSERT(wasReading || wasWriting);
+  if (wasWriting) {
+    secs = std::max(300.0, secs);
```

Comment:
Why is the timeout at least 300 seconds here? Is there any specific reason for it?

Reply:
because the idle timeout may be super low, so this is a 300 second write timeout

Reply:
If it's a different timeout here (not an idle timeout, but a write timeout), shouldn't it be configurable?